src/main/java/uk/gov/hmcts/reform/pdf/service/client/PDFServiceClient.java

```diff
@@ -83,50 +83,22 @@ public Health serviceHealthy() {
     }
 
     private URI htmlEndpoint() {
-        return pdfServiceBaseUrl.resolve(
-            String.format("/api/%s/pdf-generator%s", version, GENERATE_FROM_HTML_ENDPOINT_PATH)
-        );
+        return pdfServiceBaseUrl.resolve("/pdfs");
     }
 
-    private HttpEntity<MultiValueMap<String, Object>> requestEntityFor(
+    private HttpEntity<String> requestEntityFor(
         byte[] template,
         Map<String, Object> placeholders) {
 
-        MultiValueMap<String, Object> formData = new LinkedMultiValueMap<>();
-        formData.add("template", new FileBytesResource(template));
-        formData.add("placeholderValues", placeholders);
         HttpHeaders headers = new HttpHeaders();
-        headers.setContentType(MediaType.MULTIPART_FORM_DATA);
-        return new HttpEntity<>(formData, headers);
-    }
-
-    /**
-     * The 'filename' attribute is needed in multipart/form-data part's Content-Disposition. Otherwise
-     * the endpoint will not treat sent bytes as a MultipartFile.
-     */
-    private static class FileBytesResource extends ByteArrayResource {
-
-        private static final String DEFAULT_FILE_NAME = "template.html";
-
-        private FileBytesResource(byte[] byteArray) {
-            super(byteArray);
-        }
-
-        @Override
-        public String getFilename() {
-            return DEFAULT_FILE_NAME;
-        }
-
-        @Override
-        public boolean equals(Object obj) {
-            return super.equals(obj);
-        }
+        headers.setContentType(MediaType.valueOf("application/vnd.uk.gov.hmcts.pdf-service.v2+json"));
+        headers.setAccept(Collections.singletonList(MediaType.APPLICATION_PDF));
 
-        @Override
-        public int hashCode() {
-            return super.hashCode();
+        GeneratePdfRequest request = new GeneratePdfRequest(new String(template), placeholders);
+        try {
+            return new HttpEntity<>(OBJECT_MAPPER.writeValueAsString(request), headers);
+        } catch (JsonProcessingException e) {
+            throw new PDFServiceClientException(e);
```

Comment:
We could add a high level error message, saying we failed to convert the request to JSON. Just to consider.

Reply:
It should be picked up by `RuntimException`, thanks for pointing that out :+1:

Reply:
What I meant was that when this exception is thrown, the user of this library will see a low-level error message from JsonProcessingException, not necessarily very clear. That's why it would be beneficial to add a message saying `Failed to convert PDF request into JSON`.